CONTRIBUTING.md

```diff
@@ -0,0 +1,123 @@
+# Contributing to SPIR-V Tools
+
+
+## For users: Reporting bugs and requesting features
+
+
+We organize known future work in GitHub projects.  See [Tracking SPIRV-Tools work with GitHub projects](https://github.com/KhronosGroup/SPIRV-Tools/blob/master/projects.md) for more.
+
+
+To report a new bug or request a new feature, please file a GitHub issue. Please ensure the bug has not already been reported by searching [issues](https://github.com/KhronosGroup/SPIRV-Tools/issues) and [projects](https://github.com/KhronosGroup/SPIRV-Tools/projects). If the bug has not already been reported open a new one [here](https://github.com/KhronosGroup/SPIRV-Tools/issues/new).
+
+
+When opening a new issue for a bug, make sure you provide the following:
+
+
+
+*   A clear and descriptive title.
+    *   We want a title that will make it easy for people to remember what the issue is about.  Simply using "Segfault in spirv-opt" is not helpful because there could be (but hopefully aren't) multiple bugs with segmentation faults with different causes.
+*   A test case that exposes the bug, with the steps and commands to reproduce it.
+    *   The easier it is for a developer to reproduce the problem, the quicker a fix can be found and verified.  It will also make it easier for someone to possibly realize the bug is related to another issue.
+
+For feature requests, we use [issues](https://github.com/KhronosGroup/SPIRV-Tools/issues) as well.  Please create a new issue, as with bugs.  In the issue provide
+
+
+
+*   A description of the problem that needs to be solved.
+*   Examples that demonstrate the problem.
+
+
+## For developers: Contributing a patch
+
+Before we can use your code, you must sign the [Khronos Open Source Contributor License Agreement](https://cla-assistant.io/KhronosGroup/SPIRV-Tools) (CLA), which you can do online. The CLA is necessary mainly because you own the copyright to your changes, even after your contribution becomes part of our codebase, so we need your permission to use and distribute your code. We also need to be sure of various other things -- for instance that you'll tell us if you know that your code infringes on other people's patents. You don't have to sign the CLA until after you've submitted your code for review and a member has approved it, but you must do it before we can put your code into our codebase.
+
+See [README.md](https://github.com/KhronosGroup/SPIRV-Tools/blob/master/README.md) for instruction on how to get, build, and test the source.  Once you have made your changes:
+
+
+
+*   Create a pull request (PR) with your patch.
+*   Make sure the PR description clearly identified the problem, explains the solution, and references the issue if applicable.
+*   Watch the continuous builds to make sure they pass.
+*   Request a code review.
+
+The reviewer can either approve your PR or request changes.  If changes are requested:
+
+
+
```

Comment:
Add:

* Please make sure your code follows the Google C++ Style Guide.
* For help formatting your code, you can run the command 'clang-format -style=file -i [modified-files]'

Reply:
Fixed.